staticcheck/lint.go

```diff
@@ -117,6 +117,51 @@ func checkValidHostPort(arg int) CallCheck {
 	}
 }
 
+func checkNonOverlappingDstSrc(dstArg, srcArg int) CallCheck {
+	return func(call *Call) {
+		dst := call.Args[dstArg]
+		src := call.Args[srcArg]
+		if dst.Value == src.Value {
+			// simple case of f(b, b)
+			// add exception for f(nil, nil)
+			_, dstConst := dst.Value.Value.(*ir.Const)
```

Comment:
This will need to use irutil.Flatten on the two values, to account for sigma nodes.